### DISCUSSION TITLE
Bun v0.4.0

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#1651] [Announcements]
To install:
```bash
curl https://bun.sh/install | bash
```

To upgrade:
```bash
bun upgrade
```

**[Read Bun v0.4.0's release notes on Bun's blog](https://bun.sh/blog/bun-v0.4.0)**

<a href="https://bun.sh/blog/bun-v0.4.0"><img src="https://user-images.githubusercontent.com/3238291/209300634-79333724-1bbc-4c80-a811-23d6e1ff6373.gif" width=600 /></a>

## Commit log

- fix `__require` collision from linking by @dylan-conway in https://github.com/oven-sh/bun/pull/1585
- chore: add eslintcache by @Simon-He95 in https://github.com/oven-sh/bun/pull/1586
- Add filename completions on `bun` command to Zsh by @colinhacks in https://github.com/oven-sh/bun/pull/1593
- Exclude additional TSdeclaration file extensions from completions by @colinhacks in https://github.com/oven-sh/bun/pull/1596
- fix path string by @YUxiangLuo in https://github.com/oven-sh/bun/pull/1597
- override `process.stdin.on()` correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/1603
- fix(stream): Fix Readable.pipe() by @ThatOneBro in https://github.com/oven-sh/bun/pull/1606
- make `process.stdin` work under TTY by @alexlamsl in https://github.com/oven-sh/bun/pull/1611
- add `bun pm ls` for printing lockfiles by @dylan-conway in https://github.com/oven-sh/bun/pull/1612
- fix(stream): make Readable.read work w/o \_construct implemented by @ThatOneBro in https://github.com/oven-sh/bun/pull/1613
- Fix typo in bun.d.ts by @eltociear in https://github.com/oven-sh/bun/pull/1619
- add tests for `process.stdin` by @alexlamsl in https://github.com/oven-sh/bun/pull/1621
- docs(README.md): update bun-types new path definition by @hoseinprd in https://github.com/oven-sh/bun/pull/1622
- Delete Oniguruma by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/1625
- bug compatible with `stdin.on("readable")` by @alexlamsl in https://github.com/oven-sh/bun/pull/1626
- Implement `bunx` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/1634
- add tests for #1633 by @alexlamsl in https://github.com/oven-sh/bun/pull/1635
- fix jest hooks in bun-test by @ethanburrell in https://github.com/oven-sh/bun/pull/1639
- fix `bun install` dependency resolution by @alexlamsl in https://github.com/oven-sh/bun/pull/1643
- [install] avoid dependency conflicts between siblings by @alexlamsl in https://github.com/oven-sh/bun/pull/1647
- Update benchmarks by @colinhacks in https://github.com/oven-sh/bun/pull/1648
- [install] fix remaining corner cases with dependency resolution by @alexlamsl in https://github.com/oven-sh/bun/pull/1649

## New Contributors
* @Simon-He95 made their first contribution in https://github.com/oven-sh/bun/pull/1586
* @YUxiangLuo made their first contribution in https://github.com/oven-sh/bun/pull/1597
* @alexlamsl made their first contribution in https://github.com/oven-sh/bun/pull/1603
* @hoseinprd made their first contribution in https://github.com/oven-sh/bun/pull/1622
* @ethanburrell made their first contribution in https://github.com/oven-sh/bun/pull/1639

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.3.0...bun-v0.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.4.0'>Bun v0.4.0</a>.</em>

**Comment by FrameMuse** [comment]
We're going closer to Bun v1.0.0 😀
